src/circleci/rollcage/throwables.clj

```diff
@@ -0,0 +1,14 @@
+(ns circleci.rollcage.throwables)
+
+(defn cause-seq
+  [^Throwable ex]
+  (loop [^Throwable ex ex
+         result (list)]
+    (if-not ex
+      result
+      (recur (.getCause ex)
+             (conj result ex)))))
+
+(defn merged-ex-data
+  [^Throwable ex]
+  (reduce merge (map ex-data (cause-seq ex))))
```

Comment:
So the idea is that the merge operates over causes in reverse order? I.e. "innermost" cause first?